### PR TITLE
Support non-capturing groups (?:text)

### PR DESCRIPTION
**Pull request by Alexey-T**

### CODE REVIEW

**Comment by andgineer** [comment]
Thank you, great work!

Could you please add some test cases specific to the non-capturing work?

**Comment by Alexey-T** [comment]
Added fix and test in Lazarus.

**Comment by Alexey-T** [comment]
Screenshot from 2019-11-13 18-20-57

**Comment by andgineer** [comment]
Thank you!
But why din't you just add cases into
https://github.com/andgineer/TRegExpr/blob/master/test/tcregexp.pas

Travis (.travis.yaml) use it to automatically test the project - see check results above and as a budge in README

**Comment by Alexey-T** [comment]
Pls see again- added 2 testcases and fixed major bug in my diff.

File: src/RegExpr.pas [hunk 1]
```diff
@@ -2576,7 +2576,9 @@ function TRegExpr.ParseAtom (var flagp : integer) : PRegExprChar;
            // check for non-capturing group: (?:text)
            if (regparse + 1)^ = ':' then begin
              if regnpar < NSUBEXP then
+             begin
```
Comment (andgineer) [inline src/RegExpr.pas]: Why do we need begin/end for one liner?

File: test/tcregexp.pas [hunk 2]
```diff
@@ -273,49 +274,56 @@   TRegExTest = record
...
 
 procedure TTestRegexpr.AreEqual(ArrorMessage: string; s1,s2: string);
 begin
-  {$IFDEF FPC}
-  AssertEquals(ArrorMessage, s1,s2);
-  {$ELSE}
-  CheckEquals(s1,s2, ArrorMessage)
-  {$ENDIF}
-end;
+  {$IFDEF FPC}
```
Comment (andgineer) [inline test/tcregexp.pas]: not clear - what was changed?

File: test-non-capturing-groups/test-regex.lpi [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andgineer) [inline test-non-capturing-groups/test-regex.lpi]: Do we still need this separate test project?
Now we have the test cases in main test suit

**Comment by andgineer** [comment]
> Pls see again- added 2 testcases and fixed major bug in my diff.

Great work, thank you very much!

I just have a few minor questions about the changes

File: test-non-capturing-groups/test-regex.lpi (hunk 3 above)
Comment (Alexey-T) [reply test-non-capturing-groups/test-regex.lpi]: Removed it.

File: test/tcregexp.pas (hunk 2 above)
Comment (Alexey-T) [reply test/tcregexp.pas]: It is some EOL changes made by Lazarus, or maybe trailing spaces deleted.

**Pull request merged**